pytorch_lightning/loops/optimization/manual_loop.py

```diff
@@ -144,15 +113,10 @@ def advance(self, batch: Any, batch_idx: int) -> None:  # type: ignore[override]
 
             training_step_output = self.trainer.call_hook("training_step_end", training_step_output)
 
-            _check_training_step_output(lightning_module, training_step_output)
-
             self._hiddens = _extract_hiddens(training_step_output, lightning_module.truncated_bptt_steps)
 
             result = ManualResult.from_training_step_output(training_step_output, self.trainer.accumulate_grad_batches)
 
-            if self.trainer.terminate_on_nan:
-                check_finite_loss(result.closure_loss)
-
```

Comment:
Since the manual loop does not need to return a loss, this is no longer done. The user would need to call `check_finite_loss` manually.

This should be okay with the introduction of `detect_anomaly`